manifests/instance.pp

```diff
@@ -73,6 +78,7 @@
   $status             = $elasticsearch::status,
   $config             = undef,
   $configdir          = undef,
+  $manage_datadir     = true,
```

Comment:
Might be easier to have this one default to $elasticsearch::manage_datadir ?
Then you also don't need the logic further down.